pandas/tests/io/test_parquet.py

```diff
@@ -223,6 +224,49 @@ def test_options_get_engine(fp, pa):
         assert isinstance(get_engine("fastparquet"), FastParquetImpl)
 
 
+def test_get_engine_auto_error_message():
+    # Expect different error messages from get_engine(engine="auto")
+    # if engines aren't installed vs. are installed but bad version
+    from pandas.compat._optional import VERSIONS
+
+    # Do we have engines installed, but a bad version of them?
+    pa_min_ver = VERSIONS.get("pyarrow")
+    fp_min_ver = VERSIONS.get("fastparquet")
```

Comment:
Yeah, I thought about parametrizing, but in this case it's tricky: the error message will only ever show up if _both_ pyarrow _and_ fastparquet are inadequately installed (i.e. not installed, or bad version installed). This is embodied in the conditional:
```
if not have_usable_pa and not have_usable_fp:
```
The `and`'ed conditions can't be decoupled. So we'd still need botht he lines of code you highlight above. 

The only thing that could be de-duplicated with parametrization, is the contents of the `if` block. But even that doesn't feel quite right; it's just two aspects of the same situation. 

Maybe the best thing to do if we want to be absolutely strict, is to take all the boolean flagging that is currently in these two sections
```
    # Do we have engines installed, but a bad version of them?
[...]
    # Do we have usable engines installed?
```
and move it outside of the test function, to module level. And use `pytest.mark.skipif` (or one of those wacky fixtures that inject `pytest.mark.skip`) to only run the test when we expect an error message. 

At this point, I'm going to ask: how important is it that we do it this way? Because I'm slowly running out of steam for this.

Reply:
this test is kind of overkill , but ok.

Reply:
I see, didn't realize you need to know about both versions at the same time.

I think you could write this in a very compact way, if we create the `_HAVE_USABLE_PYARROW`... variables at the beginning of the file, like `_HAVE_PYARROW`. Then the parametrization would be trivial. But not that important.

Thanks for the work on this!

Reply:
Cool. I'm happy to come back to it later if someone sees this as adding value. Though I tend to agree with @jreback that we're bordering overkill. Anyhoo, I'm glad to have this closed before everything drops out of mental cache over Easter. 
Cheers all for the good input!!